.circleci/config.yml

```diff
@@ -0,0 +1,116 @@
+version: 2
+jobs:
+  build:
+    machine: true
+    environment:
+      PATH: "~/$CIRCLE_PROJECT_REPONAME/gradle-2.9/bin:/home/circleci/buck/bin:$PATH"
+      TERM: "dumb"
+      ADB_INSTALL_TIMEOUT: 10
+      GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx512m -XX:+HeapDumpOnOutOfMemoryError"'
+
+    working_directory: ~/$CIRCLE_PROJECT_REPONAME
+
+    branches:
+      ignore:
+        - gh-pages # list of branches to ignore
+
+    steps:
+      - checkout
+      - restore_cache:
+          key: react-native
+
+      - run:
+          command: |
+            # BUCK and android
+            if [[ ! -e /home/circleci/buck ]]; then git clone https://github.com/facebook/buck.git /home/circleci/buck; fi
+            cd /home/circleci/buck && ant
+            buck --version
+            source scripts/circle-ci-android-setup.sh && getAndroidSDK
+            buck fetch ReactAndroid/src/test/java/com/facebook/react/modules
+            buck fetch ReactAndroid/src/main/java/com/facebook/react
+            buck fetch ReactAndroid/src/main/java/com/facebook/react/shell
+            buck fetch ReactAndroid/src/test/...
+            buck fetch ReactAndroid/src/androidTest/...
+            ./gradlew :ReactAndroid:downloadBoost :ReactAndroid:downloadDoubleConversion :ReactAndroid:downloadFolly :ReactAndroid:downloadGlog :ReactAndroid:downloadJSCHeaders
+            # CIRCLE_NPM_TOKEN is in React Native project settings in Circle CI.
+            # It was generated for bestander user, easy to replace with anyone's else
+            echo "//registry.npmjs.org/:_authToken=${CIRCLE_NPM_TOKEN}" > ~/.npmrc
+            npm config set spin=false
+            npm config set progress=false
+            npm install
+            # for eslint bot
+            npm install github@0.2.4
+            cd website && npm install
+
+      - save_cache:
+          key: react-native
+          paths:
+            - "ReactAndroid/build/downloads"
```

Comment:
If the list of RN downloads changes we'll want the key to invalidate itself. Perhaps we should include the checksum of one of the Gradle files in a cache key.